File: src/dataset/python/bob_exercism-fruit-in/openAI/ChatGPT4_bob_v2.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if not hey_bob:
        return "Fine. Be that way!"
    
    question = hey_bob.endswith('?')
    yell = hey_bob.isupper()

    if question and yell:
        return "Calm down, I know what I'm doing!"
    elif question:
        return "Sure."
    elif yell:
        return "Whoa, chill out!"
    else:
        return "Whatever."

File: src/dataset/c/bob_exercism-IrvinRamiA/openAI/ChatGPT4_bob_v2.c
#include "bob.h"
#include <stdlib.h>
#include <ctype.h>
#include <stdbool.h>
#include <string.h>

char *hey_bob(const char *greeting)
{
    bool question = false;
    bool yell = true;
    bool nothing = true;

    const char *end = greeting + strlen(greeting) - 1;

    // Trim trailing spaces
    while (end >= greeting && isspace(*end))
        end--;

    if (end < greeting) // Empty or whitespace-only input
        return strdup("Fine. Be that way!");

    if (*end == '?')
        question = true;

    for (const char *c = greeting; c <= end; c++)
    {
        if (isalpha(*c))
        {
            nothing = false;
            if (islower(*c))
                yell = false;
        }
        else if (!isspace(*c))
        {
            nothing = false;
        }
    }

    if (nothing)
        return strdup("Fine. Be that way!");
    if (yell && question)
        return strdup("Calm down, I know what I'm doing!");
    if (yell)
        return strdup("Whoa, chill out!");
    if (question)
        return strdup("Sure.");
    return strdup("Whatever.");
}